src/include/sof/audio/format.h

```diff
@@ -79,6 +79,13 @@
 #define SATP_INT32(x) (((x) > INT32_MAX) ? INT32_MAX : (x))
 #define SATM_INT32(x) (((x) < INT32_MIN) ? INT32_MIN : (x))
 
+/* Arithmetic shift right */
+#define ARITHMETIC_SHIFT_RIGHT(u, n) ({	\
+	(u > 0) ?	\
+		(int32_t)((uint32_t)(((uint32_t)u) >> ((uint32_t)n))) :	\
+		(-((int32_t)((uint32_t)(((uint32_t)((int32_t)(-1 - u))) >> ((uint32_t)n))))) - 1; \
+})
```

Comment:
Honestly, holy fuck, this code looks like crap and there's no real need for it in any case.

(int)0x80000000 >> 31 == 0xFFFFFFFF

Reply:
Agreed its not pretty, but we all have different backgrounds. Some of us more mathematical and some of us more software, but that is why we work as a team to get code in. @ShriramShastry I am not sure what the point of this macro is, can you detail why there is a need for so many casts?